cpp/bench/sg/svc.cu

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/matrix/kernelparams.h>
+#include <cuml/svm/svm_model.h>
+#include <cuml/svm/svm_parameter.h>
+#include <cmath>
+#include <cuml/cuml.hpp>
+#include <cuml/svm/svc.hpp>
+#include <sstream>
+#include <utility>
+#include "benchmark.cuh"
+
+namespace ML {
+namespace Bench {
+namespace SVM {
+
+template <typename D>
+struct Params {
+  DatasetParams data;
+  BlobsParams blobs;
+  MLCommon::Matrix::KernelParams kernel;
+  ML::SVM::svmParameter svm_param;
+  ML::SVM::svmModel<D> model;
+};
+
+template <typename D>
+class SVC : public BlobsFixture<D, D> {
+ public:
+  SVC(const std::string& name, const Params<D>& p)
+    : BlobsFixture<D, D>(name, p.data, p.blobs),
+      kernel(p.kernel),
+      model(p.model),
+      svm_param(p.svm_param) {
+    //std::vector<std::string> kernel_names{"linear", "poly", "rbf", "tanh"};
+    //std::ostringstream oss;
+    //oss << name << "/" << kernel_names[kernel.kernel] << p.data;
+    //this->SetName(oss.str().c_str());
```

Comment:
Any reason for commenting this out?

Reply:
I was not sure whether I should use SetName, or pass the modified name directly to the base class. The base class also calls SetName and we overwrite that here. I guess it should be fine, so I have uncommented this.